gensim/models/word2vec.py

```diff
@@ -1204,19 +1211,45 @@ def prepare_vocab(self, hs, negative, wv, update=False, keep_raw_vocab=False, tr
         sample = sample or self.sample
         drop_total = drop_unique = 0
 
+        # if max_vocab is specified instead of min_count
+        # pick a min_count which satisfies max_vocab as well as possible
+        if self.max_vocab is not None:
+            sorted_vocab = sorted(self.raw_vocab.keys(), key=lambda word: self.raw_vocab[word], reverse=True)
+
+            if self.max_vocab < len(sorted_vocab):
+                calc_min_count = self.raw_vocab[sorted_vocab[self.max_vocab]] + 1
+            else:
+                calc_min_count = 1
+
+            if calc_min_count > min_count:
+                logger.info("effective_min_count was set to %d due to max_vocab being set to %d" %
+                           (calc_min_count, self.max_vocab))
+                effective_min_count = calc_min_count
+            else:
+                logger.info("""specified min_count = %d is larger that min_count calculated
+                               by max_vocab = %d, using specified min_count""" % (min_count, calc_min_count))
+                effective_min_count = None
```

Comment:
To remain consistent with its name, `effective_min_count`, this variable should be the value that's actually used - never `None`, always the `max(min_count, calc_min_count)`.